### PR TITLE
step 0.15.7

### PR DESCRIPTION
**Pull request by dopey**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/step.rb [hunk 1]
```diff
@@ -16,8 +16,8 @@ class Step < Formula
   depends_on "go" => :build
 
   resource "certificates" do
-    url "https://github.com/smallstep/certificates/releases/download/v0.15.5/step-certificates_0.15.5.tar.gz"
-    sha256 "861690259f21d60e1491f28fd572805b1f8c3141bde651e018da06b404d7cfed"
+    url "https://github.com/smallstep/certificates/releases/download/v0.15.8/step-certificates_0.15.8.tar.gz"
```
Comment (SMillerDev) [inline Formula/step.rb:19]: These are no longer in sync?

Comment (dopey) [reply Formula/step.rb:19]: I assume you're referring to the versions for `cli` and `certificates`? If so, then yes (although, they were already out of sync). They start off in sync at the beginning of a minor release (e.g. v0.15.0). But then we often have to make more changes `certificates` than we do to the `cli`. We'll often add new features that don't immediately require `cli` parity.

Separating these two from the same homebrew package is on our roadmap. We just need `cli` to depend on `certificates`. But we haven't gotten around to it yet. Hopefully soon.

**Review by SMillerDev — APPROVED** [review]
Thanks @dopey  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by tashian** [comment]
Hi @SMillerDev, I work with @dopey on step. How do we go about getting an ARM64 bottle for this formula?

**Comment by carlocab** [comment]
Does step support ARM yet? If it does, I can dispatch a bottle job for it.

**Comment by tashian** [comment]
@carlocab It does. Will it be an issue that step formula depends on `dep`, which doesn't have an arm build? We don't really depend on `dep` anymore, we can remove if needed.

**Comment by carlocab** [comment]
Yes, it needs to be removed. Please open a PR doing that. Tag me when you do so, as I need to add a label that will force the ARM node to attempt a build.

**Comment by dopey** [comment]
@carlocab https://github.com/Homebrew/homebrew-core/pull/72255. Let me know if this looks ok. Thanks!

**Comment by dopey** [comment]
Hey @carlocab, was wondering if I could get your expertise on a bug we've recently encountered with our homebrew package.

In the last two homebrew-core releases our `step version` has begun returning a different value. An example of the current return is:

```
$> /usr/local/bin/step version
Smallstep CLI/refs/pull/71442/merge (darwin/amd64)
Release Date: 2021-02-18 22:18 UTC
```

Where in the past it would say `Smallstep 0.15.8` (for example). Our formula uses `make` to compile a binary, and the target consumes a `$VERSION` environment variable that is then baked into the executable. 

We examined a few other formulas to see if we could find any hints as to what might be happening. One "hint" is this use of the value `#{version}` (https://github.com/Homebrew/homebrew-core/blob/master/Formula/vegeta.rb#L23) in the vegata formula. We're wondering if maybe homebrew-core is setting an `VERSION` in the environment before running the build steps? 

One reason that we're not sure about homebrew-core setting the `VERSION` environment variable is that we're building two binaries in our formula. The second binary is built in a `resource.stage` block (https://github.com/Homebrew/homebrew-core/blob/master/Formula/step.rb#L28-L31) so maybe the `VERSION` variable is not applied to subblocks?

Assuming this does indeed have to do with name collision on the `VERSION` keyword, that should be easy for us to fix. We just want to make sure we know what the issue is so we can avoid the conflict in the future.

Thanks!

**Comment by carlocab** [comment]
AFAIK `brew` doesn't set `VERSION` during the build. When is `VERSION` supposed to be set? Is it defined in your `Makefile`(s)?

You can have a look at the environment during the build by adding `system "env"` calls in the `install` method and doing `brew install -sv`. You can also always just add the right version to the environment with `ENV["VERSION"] = version.to_s`.

**Pull request closed without merging**